File: 338_CountingBits.cpp
//60.08%
class Solution {
public:
    vector<int> countBits(int num) {
        if(num == 0)
			return {0};
		vector<int> res(num + 1);
		for(int i = 1; i <= num; ++i){
			int l(i);
			//++res[i];
			while(l){
				++res[i];
				l &= (l - 1);
			}
		}
		return res;
    }
};

File: 338_CountingBits.py
##29.55%
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        if 0 == num:
            return [0]
        res = [0] * (num + 1)
        for i in xrange(1, num + 1):
            l = i
            while l != 0:
                res[i] += 1
                l &= (l - 1)
        return res